manifests/candlepin.pp

```diff
@@ -128,15 +132,26 @@
       key_group   => $group,
       key_mode    => '0440',
     } ~>
-    exec { 'import CA into Candlepin truststore':
-      command => "keytool -import -trustcacerts -v -keystore ${keystore} -storepass ${keystore_password} -alias ${alias} -file ${ca_cert} -noprompt",
-      unless  => "keytool -list -keystore ${keystore} -storepass ${keystore_password} -alias ${alias}",
+    file { $truststore_password_path:
+      ensure  => file,
+      content => $truststore_password,
+      owner   => $user,
+      group   => $group,
+      mode    => '0440',
+    } ~>
+    exec { 'Create Candlepin truststore with CA':
+      command => "keytool -import -v -keystore ${truststore} -alias ${alias} -file ${ca_cert} -noprompt -storetype pkcs12 -storepass ${truststore_password}",
+      unless  => "keytool -list -keystore ${truststore} -alias ${alias} -storepass ${truststore_password}",
```

Comment:
This was already a problem before, but passing it in as a command line option means it's visible in `ps aux`. Can keytool read `$truststore_password_path` like openssl does with `file:<path>`?

Reply:
There is and I fixed that everywhere